### PR TITLE
[IMP] mail: add a check on inverse relation name in entities

### PR DESCRIPTION
**Pull request by xdu-odoo**

### CODE REVIEW

File: addons/mail/static/src/messaging/entity/entity_core.js [hunk 1]
```diff
@@ -65,6 +65,19 @@ function _checkFields(Entities) {
                     }" has no inverse field "${RelatedEntity.name}/${field.inverse}".`
                 );
             }
+            if (inverseField.inverse !== fieldName) {
+                throw new Error(
+                    `Relation "${
+                        Entity.name
+                    }/${
+                        fieldName
+                    }" field name do not match with inverse field inverse name definition "${
```
Comment (alexkuhn) [inline addons/mail/static/src/messaging/entity/entity_core.js]: _Relation $1/$2 field name do not match with inverse field inverse name definition $3/$4._
=>
_Inverse field name of relation $1/$2 does not match with field name of relation $1/$2._

I think the sentence is clearer that way, and gives more clue to how to fix it.

**Review by alexkuhn — CHANGES_REQUESTED** [review]
I would also make advantage of this PR to add some checks on `to` (i.e. name of entity class).

File: addons/mail/static/src/messaging/entity/entity_core.js (hunk 1 above)
Comment (xdu-odoo) [reply addons/mail/static/src/messaging/entity/entity_core.js]: Ahah thanks, I didn't find a way to be clear :p

**Comment by xdu-odoo** [comment]
> I would also make advantage of this PR to add some checks on `to` (i.e. name of entity class).

In fact, I tried but another is generated before (it does not find the inverse field).
I tried by changing in `thread.js` the ThreadCache relation to point on Composer or another entity and I got the "has no inverse field" error message. So I would say that the case is already handled, what do you think ?

**Comment by alexkuhn** [comment]
Checking `inverse` and `to` should be done at the same time. The combination of both pieces are checked and when something is wrong, it's hard to know whether it's because of typo in `inverse` or `to` (or maybe even in both!).

**Comment by xdu-odoo** [comment]
> Checking `inverse` and `to` should be done at the same time. The combination of both pieces are checked and when something is wrong, it's hard to know whether it's because of typo in `inverse` or `to` (or maybe even in both!).

Yeah I understand but if you misconfigured in one side you can't find an inverse field so you get the inverse field missing error (which is the case so IMO it's correct) and if you misconfigured in the other side you get the error saying the relation is misconfigured as inversefield.to should be a given class or a parent.

So I don't know what check could we add here :o

**Review by alexkuhn — APPROVED** [review]
LGTM.

Note that this is not really as worthy as it seems, because with entity fields branch I added complex stuffs in field normalization step and this is usually where problems arise from bad field declaration. Check on fields should happen before field normalization, and inverse is optional in declared fields.

Still nice to keep field checks after normalization step, just to ensure the produced field definitions make sense.

_(BTW, "normalization" is a bad choice of word, gotta find something better...)_

**Comment by alexkuhn** [comment]
Oh, and don't forget task-id, or you will get your wrist slapped by seb...

**Pull request merged**